packages/server/utils/PubSubPromise.ts

```diff
@@ -37,12 +39,27 @@ export default class PubSubPromise<Request, Response> {
   }
 
   publish = (request: Request) => {
-    return new Promise<Response>((resolve, reject) => {
+    return new Promise<Response>((resolve) => {
       const nextJob = numToBase64(this.jobCounter++)
       const jobId = `${SERVER_ID}:${nextJob}`
       const timeoutId = setTimeout(() => {
         delete this.jobs[jobId]
-        reject(new Error('Redis took too long to respond'))
+        const errorMessage = `GQL Executor took more than ${MAX_TIMEOUT}ms to respond`
+        const {
+          authToken,
+          query = '',
+          variables = '',
+          docId = ''
+        } = (request as unknown) as GQLRequest
```

Comment:
In [getGraphQLExecutor](https://github.com/ParabolInc/parabol/blob/bfed594b1f8f053ea628a3c541502ddd9b47b374/packages/server/utils/getGraphQLExecutor.ts#L6), we specify that the first arg should be `GQLRequest` but the request here is unknown as it could theoretically be anything. I'm sure there's a better way that I'm forgetting but I've also spent more time than I prob should on this before calling for help. Do you have any suggestions for how we can destructure the req properties without casting?

Reply:
ahhh this is why i love typescript! i definitely did you a disservice pointing you to this file in the issue.

the quick & dirty is to stringify the whole request `tags: {payload: JSON.stringify(request)}` and hope that it doesn't exceed 255 chars

the _right_ way is to move the handling of the error to the correct layer.

Basically we've got this "transport" layer that uses redis & only cares about redis pub/sub stuff. it doesn't care what it sends, just give it a channel & it'll send it.

Next layer up is the graphql or "data transport" layer. if tomorrow we replace redis with rabbitmq, aws, etc, it wouldn't change a thing. all it cares about is the gql fields. You correctly found this to be `getGraphQLExecutor`.

To fix, it'd look like this:
- put the `reject` back in, maybe `reject('TIMEOUT')`
- in the try/catch of `handleGraphQLTrebuchetRequest` go into the `catch` statement
- check if e?.message === 'TIMEOUT', if so, then send the fields you want to sentry & return the standard graphql response `{data: null, errors:[{message: 'The request took too long'}]}`

Reply:
Ah thanks, that makes sense. Updated! 

Previously, I changed the timeout limit to 100ms instead of 10000ms and I could see that new reflections were disappearing and the fatal error message would appear, particularly at 10/20 ms timeout. When I added the return statement, the user would see the loading spinner instead of an error message. Strangely, now when I try to reproduce that behaviour, when I do or don't include the return statement in `PubSubPromise` or `handleGraphQLTrebuchetRequest`, I don't see any difference in behaviour. It typically just shows the loading spinner. Still, I was able to repro the fix previously and this change seems safe to try.